src/test/e2e/21_connect_creds_test.go

```diff
@@ -71,10 +71,12 @@ func TestMetrics(t *testing.T) {
 	client := &http.Client{Transport: tr}
 	httpsEndpoint := strings.ReplaceAll(tunnel.HTTPEndpoint(), "http", "https")
 	resp, err := client.Get(httpsEndpoint + "/metrics")
+	defer func() {
+		require.NoError(t, resp.Body.Close())
```

Comment:
This should be after the error, as if it's before we can get a nil pointer exception on `resp.Body`

Reply:
Agreed, it's unlikely but best for this to be watertight.